### ISSUE TITLE
LightGBM predicts negative values

### ISSUE THREAD

**Issue by seyidcemkarakas** [microsoft/LightGBM#5629]
Hi all !

My **LightGBM regressor** model returns negative values.
For **XGBoost** there is `objective='count:poisson'` hyperparameter in order to prevent returning negative predicitons.

Is there any chance to do this ?

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

LightGBM also supports poisson regression. For example, consider the following, based on `scikit-learn`'s "Tweedie regression for on insurance claims" tutorial ([link](https://scikit-learn.org/stable/auto_examples/linear_model/plot_tweedie_regression_insurance_claims.html)).

```python
import lightgbm as lgb
import numpy as np
from matplotlib import pyplot

# random Poisson-distributed target and one informative feature
y = np.random.poisson(lam=15.0, size=1_000)
X = y + np.random.normal(loc=10.0, scale=2.0, size=(y.shape[0], ))
X = X.reshape(-1, 1)

# fit a Poisson regression model
reg = lgb.LGBMRegressor(
    objective="poisson",
    n_estimators=150,
    min_data=1
)
reg.fit(X, y)

# get predictions
preds = reg.predict(X)

print("summary of predicted values")
print(f"  * min: {round(np.min(preds), 3)}")
print(f"  * max: {round(np.max(preds), 3)}")

# compare predicted distribution to the empirical one
bins = np.linspace(0, 30, 50)
pyplot.hist(y, bins, alpha=0.5, label='actual')
pyplot.hist(preds, bins, alpha=0.5, label='predicted')
pyplot.legend(loc='upper right')
pyplot.show()
```

<img width="527" alt="image" src="https://user-images.githubusercontent.com/7608904/207499926-0953ce3e-d4e6-4746-a3fc-d50fecaba345.png">

However... I don't recommend using Poisson regression just to achieve "no negative predictions". The Poisson loss function is intended to be used for cases where you believe your target is poisson-distributed, e.g. it looks like counts of events observed over some regularly interval like time or space.

Other options you might consider to try to achieve the behavior "never predict a negative number from LightGBM regression":

* write a custom objective function in one of the interfaces that support it, like the R or Python package
* post-process LightGBM's predictions, recoding negative values to 0
* pre-process the target variable such that there are no negative values (e.g. dropping such observations, re-scaling, taking the absolute value)

**Comment by jameslamb** [comment]
On a related note...I see that you chose to double-post this here and on Stack Overflow ([link](https://stackoverflow.com/questions/74784151/lightgbm-predicts-negative-values)).

Please do not do that. Maintainers here also monitor the `[lightgbm]` tag on Stack Overflow. I could have been spending time preparing an answer here while another maintainer was spending time answering your Stack Overflow post, which would have been a waste of maintainers' limited attention that could otherwise have been spent improving this project. Double-posting also makes it less likely that others with a similar question will find the relevant discussion and answer.

I've posted my answer here onto that Stack Overflow post. If you agree that this answers your question, please indicate that over on Stack Overflow.
